src/components/node.js

```diff
@@ -87,6 +88,8 @@ class Node extends React.Component {
    */
 
   shouldComponentUpdate = (nextProps) => {
+    if (nextProps.readOnly !== this.props.readOnly) return true
```

Comment:
I think this line should go below the one a few down that says: `Component && Component.suppressShouldComponentUpdate`, since that one should be able to override all `shouldComponentUpdate` logic.

Could you add a comment above it of something like:

```js
// If the `readOnly` status has changed, we need to re-render in case there is
// any user-land logic that depends on it, like nested editors.
```